lua/core/mappings.lua

```diff
@@ -2,14 +2,117 @@ local is_available = astronvim.is_available
+  else
+    vim.go.background = "light"
+  end
+  print(string.format("background=%s", vim.go.background))
+end
+
+-- Toggle completion (by running cmp setup again).
+-- local toggle_completion = function()
+--   astrovim.cmp_enable = not astronvim.cmp_enable
+--   print(string.format("completion=%s", bool2str(astrovim.cmp_enable)))
+-- end
+
+-- Toggle signcolumn="auto"|"no"
+local toggle_signcolumn = function()
+  local signcolumn = vim.wo.signcolumn  -- local to window
+  if signcolumn == "no" then
+    vim.wo.signcolumn = "auto"
+  else
+    vim.wo.signcolumn = "no"
+  end
+  print(string.format("signcolumn=%s", vim.wo.signcolumn))
+end
+
+-- Set the indent and tab related numbers.
+local set_indent = function()
+  local indent = tonumber(
+    vim.fn.input("Set indent value (>0 expandtab, <=0 noexpandtab, 0 vim defaults): ")
+  )
+  if not indent then
+    indent = 0
+  end
+  vim.bo.expandtab = (indent > 0)       -- local to buffer
+  indent = math.abs(indent)
+  vim.bo.tabstop = indent               -- local to buffer
+  vim.bo.softtabstop = indent           -- local to buffer
+  vim.bo.shiftwidth = indent            -- local to buffer
+  print(string.format("indent=%d %s", indent, vim.bo.expandtab and "expandtab" or "noexpandtab"))
+end
+
+-- Change the number display modes.
+local change_number = function()
+  local number = vim.wo.number          -- local to window
+  local relativenumber = vim.wo.relativenumber  -- local to window
+  if (number == false) and (relativenumber == false) then
+    vim.wo.number = true
+    vim.wo.relativenumber = false
+  elseif (number == true) and (relativenumber == false) then
+    vim.wo.number = false
+    vim.wo.relativenumber = true
+  elseif (number == false) and (relativenumber == true) then
+    vim.wo.number = true
+    vim.wo.relativenumber = true
+  else -- (number == true) and (relativenumber == true) then
+    vim.wo.number = false
+    vim.wo.relativenumber = false
+  end
+  print(
+    string.format("number=%s, relativenumber=%s",
+      bool2str(vim.wo.number),
+      bool2str(vim.wo.relativenumber)
+    )
+  )
+end
+
+-- Toggle spell.
+local toggle_spell = function()
+  vim.wo.spell = not vim.wo.spell       -- local to window
+  print(string.format("spell=%s", bool2str(vim.wo.spell)))
+end
+
+-- Toggle syntax/treesitter
+local change_syntax = function()
+  local parsers = require("nvim-treesitter.parsers")
+  if parsers.has_parser() then
+    if vim.g.syntax_on then               -- local to buffer
+      vim.cmd("TSBufDisable highlight")
+      vim.cmd("syntax off")
+    else
+      vim.cmd("TSBufEnable highlight")
+      vim.cmd("syntax on")
+    end
+    local state = bool2str(vim.g.syntax_on)
+    print(string.format("treesitter %s", state))
+  else
+    if vim.g.syntax_on then
+      vim.cmd("syntax off")
+    else
+      vim.cmd("syntax on")
+    end
+    local state = bool2str(vim.g.syntax_on)
+    print(string.format("syntax %s", state))
+  end
+end
```

Comment:
Can you please move these to `utils.lua`

Reply:
Sure.  with astrovim